node/chainSimulator/components/coreComponents_test.go

```diff
@@ -140,13 +140,22 @@ func createArgsCoreComponentsHolder() ArgsCoreComponentsHolder {
 		ConsensusGroupSize:          1,
 		MetaChainConsensusGroupSize: 1,
 		RoundDurationInMs:           6000,
-		CreateGenesisNodesSetup: func(nodesFilePath string, addressPubkeyConverter core.PubkeyConverter, validatorPubkeyConverter core.PubkeyConverter, genesisMaxNumShards uint32) (sharding.GenesisNodesSetupHandler, error) {
-			return sharding.NewNodesSetup(nodesFilePath, addressPubkeyConverter, validatorPubkeyConverter, genesisMaxNumShards)
-		},
-		CreateRatingsData: func(arg rating.RatingsDataArg) (process.RatingsInfoHandler, error) {
-			return rating.NewRatingsData(arg)
-		},
+		RunTypeCoreComponents:       runTypeCoreComponents,
+	}
+}
+
+func createRunTypeCoreComponents() (mainFactory.RunTypeCoreComponentsHolder, error) {
```

Comment:
This is copy-pasted with return error. In test files, we could remove the error, perhaps only have `t *tesing.T` as input param